rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -775,19 +793,24 @@ def _collect_features(
             bilou_tagging=self.component_config[BILOU_FLAG],
             type=(
                 None
-                if self._uses_sequence_features_for_text()
+                if self._uses_sequence_features_for_input_text()
                 else FEATURE_TYPE_SENTENCE
             ),
         )
-        if self._uses_sequence_features_for_text() and all(
-            not feats[TEXT] for feats in features
+
+        # NOTE: This check is a last santiy check but it doesn't solve every problem.
+        # We need the list of featurizers for this component and need to raise an
+        # error if LexicalSyntacticFeaturizer is included on it's own - or together
+        # with other featurizers. The latter is not checked by the following. Otherwise,
+        # we'll run into problems (especially if entity recognition is enabled).
```

Comment:
Not a helpful comment. We should either write what problem we run into or remove this part of the comment.